cpp/src/slice2cpp/Gen.cpp

```diff
@@ -1152,6 +1155,28 @@ Slice::Gen::ForwardDeclVisitor::visitEnum(const EnumPtr& p)
         }
     }
     H << eb << ';' << sp;
+
+    H << nl << "::std::ostream& operator<<(::std::ostream&, " << mappedName << " value);";
+
+    // Generate an overload of 'operator<<' for this enum.
+    // If the provided value corresponds to a named enumerator value, we print the corresponding name.
+    // Otherwise, we print the underlying integer value.
+    C << sp;
+    C << nl << "::std::ostream&";
+    C << nl << p->mappedScope().substr(2) << "operator<<(::std::ostream& os, " << mappedName << " value)";
+    C << sb;
+    C << nl << "switch (value)";
+    C << sb;
+    for (const auto& enumerator : enumerators)
+    {
+        const string enumeratorName = enumerator->mappedName();
+        C << nl << "case " << mappedName << "::" << enumeratorName << ":";
+        C << nl << "    return os << \"" + enumeratorName + "\";";
+    }
+    C << nl << "default:";
+    C << nl << "    return os << static_cast<" << underlying << ">(value);";
```

Comment:
It's better to cast it to an int (or std::int32_t). `operator<<` for uint8_t is not what you expect.

Reply:
I casted it to `std::int32_t`.